src/database.ts

```diff
@@ -3635,8 +3675,14 @@ class Database extends common.GrpcServiceObject {
           span.addEvent('No session available', {
             'session.id': session?.id,
           });
-          this.writeAtLeastOnce(mutations, options, cb!);
-          span.end();
+          // Retry this method.
+          this.writeAtLeastOnce(mutations, options, (err, resp) => {
+            if (err) {
+              setSpanError(span, err);
+            }
+            span.end();
+            cb!(err, resp);
+          });
```

Comment:
@odeke-em this will work fine with this.writeAtLeastOnce(mutations, options, cb!);
          span.end();
  why are we making any changes here?
  
  @surbhigarg92 any thoughts?